### ISSUE TITLE
[FEATURE] Add select helpers `all_repeating()` and `all_nonrepeating()` to select forms by instrument type

### ISSUE THREAD

**Issue by ezraporter** [CHOP-CGTInformatics/REDCapTidieR#113]
# Feature Request Description
These functions would work with `extract_tibbles()` and eventually `extract_tibble()` to make it easy to select all forms of a certain type.

Suggested usage:

```r
extract_tibbles(supertbl, tbls = all_repeating())

extract_tibbles(supertbl, tbls = all_repeating() & starts_with("infusion_"))
```

# Proposed Solution
We may need to do some additional digging to figure out how to implement this. Currently we implement `tidyselect` functionality by creating a `tibble` where `redcap_form_name`s are column names and applying `tidyselect` functions to that `tibble`. That would no longer work if we wanted selectors that take into account form attributes besides the names (such as types).

I found these helpful for thinking a bit about the issue:

- [`tidyselect::peek_vars()` and `tidyselect::peek_data()`](https://github.com/r-lib/tidyselect/blob/3d538d5f01e6f41807512ed39cd5173105735fd9/R/vars.R)
- Implementation of basic [`tidyselect` helpers](https://github.com/r-lib/tidyselect/blob/b449033dde6800ae8e098fab132ee63822c09e8d/R/helpers-pattern.R)
- Implementation of [`recipes` selectors](https://github.com/tidymodels/recipes/blob/e531656dfaa87b71ff6273ddcb3412e7c040e0fc/R/selections.R) which is modeled after `tidyselect` but for recipes rather than `tibble`s

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [ ] Developers are assigned to the issue
- [x] Labels are assigned to the issue

**Comment by skadauke** [comment]
I know that this would be yet another API change, but I think the following usage for tidyselect-enabled functions would be more consistent with `dplyr::select()` and more elegant:

```r
extract_tibbles(supertbl, ...)
```

Where `...` is one or more tidyselect expressions.

Example usage would then be:

```r
supertbl |>
   extract_tibbles(all_nonrepeating(), some_repeated_instrument)
```

**Comment by ezraporter** [comment]
Right, that does make sense. We could also still support the `tbls` argument so it's not a breaking change.
